torch/_inductor/ir.py

```diff
@@ -2549,6 +2549,15 @@ def create(cls, inputs, dim):
         new_size = list(inputs[0].get_size())
         offsets_start = [0]
         offsets_end = [new_size[dim]]
+
+        # special case for cat'ing with an empty tensor -
+        # just drop the 'empty' inputs so they don't confuse the logic below.
+        def non_empty_input(input):
+            input_size = input.get_size()
+            return len(input.get_size()) > 1 or input.get_size()[0] > 0
```

Comment:
right now the assumption is that if a size is 0, it will always be guarded on.. i dont know how much we should be thinking about unbacked symints in the future.

Reply:
which decomp would that be, there is a decomp for cat?  if so how come we are lowering cat into inductor?

Reply:
https://github.com/pytorch/pytorch/blob/main/torch/_inductor/decomposition.py#L209-L212. It just exists to simplify one case.


Could extend the decomp so that if there are no 0-dim inputs, return NotImplemented. Otherwise, call cat recursively with those elements removed.

Reply:
ok- I pushed a change to move the special case out of ir.py and into decomps.  However, I couldn't delete the special case from split_cat since iiuc its a pre-autograd pass.  Please lmk if this version is good to land